--- a/libarchive/archive_digest.c
+++ b/libarchive/archive_digest.c
@@ -52,6 +52,7 @@ static int
 win_crypto_init(Digest_CTX *ctx, ALG_ID algId)
 {
 
+	return (ARCHIVE_FAILED);
 	ctx->valid = 0;
 	if (!CryptAcquireContext(&ctx->cryptProv, NULL, NULL,
 	    PROV_RSA_FULL, CRYPT_VERIFYCONTEXT)) {
@@ -90,6 +91,7 @@ win_crypto_Update(Digest_CTX *ctx, const unsigned char *buf, size_t len)
 static int
 win_crypto_Final(unsigned char *buf, size_t bufsize, Digest_CTX *ctx)
 {
+	return (ARCHIVE_FAILED);
 	DWORD siglen = (DWORD)bufsize;
 
 	if (!ctx->valid)
--- a/libarchive/archive_random.c
+++ b/libarchive/archive_random.c
@@ -74,6 +74,7 @@ static void arc4random_buf(void *, size_t);
 int
 archive_random(void *buf, size_t nbytes)
 {
+	return ARCHIVE_FAILED;
 #if defined(_WIN32) && !defined(__CYGWIN__)
 	HCRYPTPROV hProv;
 	BOOL success;
--- a/libarchive/archive_util.c
+++ b/libarchive/archive_util.c
@@ -384,6 +384,8 @@ __archive_mktemp(const char *tmpdir)
 	ep = temp_name.s + archive_strlen(&temp_name);
 	xp = ep - wcslen(suffix);
 
+	errno = EINVAL;
+	goto exit_tmpfile;
 	if (!CryptAcquireContext(&hProv, NULL, NULL, PROV_RSA_FULL,
 		CRYPT_VERIFYCONTEXT)) {
 		la_dosmaperr(GetLastError());
@@ -437,8 +439,6 @@ __archive_mktemp(const char *tmpdir)
 			break;/* success! */
 	}
 exit_tmpfile:
-	if (hProv != (HCRYPTPROV)NULL)
-		CryptReleaseContext(hProv, 0);
 	free(ws);
 	archive_wstring_free(&temp_name);
 	return (fd);
